packages/dds/tree/src/test/feature-libraries/flex-tree/lazyNode.spec.ts

```diff
@@ -444,7 +444,7 @@ describe("LazyNode", () => {
 
 		it("boxing", () => {
 			assert.equal(node.foo, node.boxedFoo.content);
-			assert(node.bar.isSameAs(node.boxedBar));
```

Comment:
If these two tests are the only use of this API, I suppose we can remove it. We still don't guarantee stable object identity of fields, but it works well enough for these cases, and nothing else cares apparently. Maybe just movbed the notes from isSameAs to the FlexTreeField type as a disclaimer.

Reply:
Well, it's also used in arrayNode.ts and lazyField.ts, above. But maybe those usages are unnecessary?

Reply:
I think they are unnecessary. In fact, even before the removal of `isSameAs`, I don't see a necessary reason for the previous logic above. If the fields are logically the same, then it shouldn't matter which one is used for the edit.